ComponentKit/Core/CKComponentController.mm

```diff
@@ -79,6 +79,56 @@ @implementation CKComponentController
   CKAppliedComponentAnimationMap _appliedAnimations;
 }
 
+static NSArray *ClassGetSubclasses(Class parentClass)
+{
+  int numClasses = objc_getClassList(NULL, 0);
+  Class *classes = NULL;
+
+  classes = (__unsafe_unretained Class *)malloc(sizeof(Class) * numClasses);
+  numClasses = objc_getClassList(classes, numClasses);
+
+  NSMutableArray *result = [NSMutableArray array];
+  for (NSInteger i = 0; i < numClasses; i++)
+  {
+    Class superClass = classes[i];
+    do
+    {
+      superClass = class_getSuperclass(superClass);
+    } while(superClass && superClass != parentClass);
+
+    if (superClass == nil)
+    {
+      continue;
+    }
+
+    [result addObject:classes[i]];
+  }
+
+  free(classes);
+
+  return result;
+}
+
++ (void)load {
+  NSMutableArray *badClasses = [NSMutableArray array];
+  for (Class klass in ClassGetSubclasses(self)) {
+    if ([klass instanceMethodForSelector:@selector(init)] !=
+        [[CKComponentController class] instanceMethodForSelector:@selector(init)]) {
+      [badClasses addObject:klass];
+    }
+  }
+  FBFatal(badClasses.count == 0, @"found bad component controllers:%@", badClasses);
+}
```

Comment:
lol, gotta remove this. Hopefully other apps have designated initializer warnings turned on.